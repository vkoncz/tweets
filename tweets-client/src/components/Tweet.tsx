import moment from 'moment';
import React from 'react';
import { TweetModel } from '../models/tweet.model';
import {
  TweetContainer,
  TweetContent,
  TweetDate,
  TweetHeader,
  UserId,
  UserName,
} from './Tweet.styles';

const Tweet: React.FC<TweetModel> = ({ userName, userId, date, content }) => {
  return (
    <TweetContainer>
      <TweetHeader>
        <UserName>{userName}</UserName> <UserId>{userId}</UserId>
        <TweetDate> Â· {moment(date).fromNow()}</TweetDate>
      </TweetHeader>
      <TweetContent>{content}</TweetContent>
    </TweetContainer>
  );
};

export default Tweet;
