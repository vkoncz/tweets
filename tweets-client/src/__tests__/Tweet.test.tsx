import React from 'react';
import { render } from '@testing-library/react';
import { TweetModel } from '../models/tweet.model';
import Tweet from '../components/Tweet';
export const tweetMock: TweetModel = {
  _id: 'test-id',
  content: 'test-content',
  date: '2019-12-31T23:00:00.000Z',
  userId: 'test-user-id',
  userName: 'test-user-name',
};

test('Tweet snapshot', () => {
  const { container } = render(<Tweet {...tweetMock} />);

  expect(container).toMatchSnapshot();
});
