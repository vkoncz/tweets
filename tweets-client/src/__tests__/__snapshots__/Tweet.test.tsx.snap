// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tweet snapshot 1`] = `
<div>
  <article
    class="sc-bdfBwQ iWRkbk"
  >
    <p
      class="sc-gsTCUz crUlbz"
    >
      <span
        class="sc-dlfnbm fXvGP"
      >
        test-user-name
      </span>
       
      <span
        class="sc-hKgILt dQuplh"
      >
        test-user-id
      </span>
      <span
        class="sc-hKgILt sc-eCssSg dQuplh"
      >
         Â· 
        a year ago
      </span>
    </p>
    <p
      class="sc-jSgupP imfxUq"
    >
      test-content
    </p>
  </article>
</div>
`;
